rfcs/min-relations.md

```diff
@@ -0,0 +1,644 @@
+
+These advantages make even `min-relations` a compelling alternative to both the Entity-in-component and external graph data structure patterns if:
+
+1. You don't care about performance.
+2. You don't need to perform complex traversals.
+3. You're operating on general graphs that can contain cycles and self-referential edges.
+4. You don't want to worry about synchronization bugs.
+
+Let's take a look at how you could use relations to build an API for a tree-shaped graph that looks suspiciously like Bevy 0.5's parent-child hierarchy.
+
+## Reference-level explanation
+
+TODO: Boxy explains the magic.
+
+### Data Storage Model
+
+### The Second Relation Filter Type Parameter
+
+## Drawbacks
+
+1. Introduces another core abstraction to users that they will have to learn.
+Rewriting our parent-child code in terms of relations is important, but will introduce the concept to users quite quickly.
+2. Heavy use of relations can seriously fragment archetypes. We need to be careful that this doesn't badly impact our performance.
+3. `move_component`, `move_relation` and `change_target` only work on `Clone` components / relations due to the need to temporarily be able to access removed component data.
+
+## Rationale and alternatives
+
+### Why do relations fragment archetypes?
+
+TODO: Boxy explains her decisions.
+
+### Why are relations stored in the ECS?
+
+TODO: Boxy help!
+
+### Why aren't relations described as a special type of components?
+
+They don't behave in exactly the same ways,
+and describing them as such is confusing to beginners.
+
+Under the hood, the existing implementation models
+
+TODO: add examples of the differences
+
+## Why do we need the builder pattern for filtering relations?
+
+TODO: Boxy explains perf implications
+
+### Why don't we just store Entities in components?
+
+See [Motivation].
+
+### Why don't we just store graph data structures in a Resource?
+
+See [Motivation] *and* [Entity Graphs].
+
+### Why do we need arbitrarily complex relation filters??
+
+Relation filters take advantage of archetypes to be much faster and more ergonomic than doing the same operation by naive iteration.
+In order to ensure that it can always be used effectively, we need a genuinely complete DSL,
+in addition to the convenient methods discussed first.
+
+The API for compound relation filters is simple to implement and easily ignored.
+
+### Why is my favorite feature relegated to Future Work?
+
+It's already many thousands of words long, and the associated PR has thousands of lines of code.
+Done is better than perfect, and we had to draw the line somewhere.
+As with all things, we can gradually improve this over time.
+
+The set of features expressed here is useful enough to be a clear improvement over the existing patterns in at least some use cases.
+There's a lot of value in getting this feature in front of users to see the response at scale.
+
+### Why is the implementation for this so *slow*?
```

Comment:
It could be helpful to provide some measure of how slow this is, and what operations are particularly bad.

- Are queries fast? 
- Is it only adding/changing relations that is slow?

A starting point could be to run the ECS benchmark as-is with some relation filters on to see if there is a tangible impact. Perhaps seeing how adding/removing relations compares to component add/removes. Having *some* information, even if it isn't complete or perfect, would help users understand the order of magnitude they should expect.

> Many uses of relations are not performance-critical,

I only say this because "performance-critical" is a hugely relative metric.